packages/commonwealth/client/scripts/state/ui/thread/index.ts

```diff
@@ -0,0 +1,35 @@
+import { createBoundedUseStore } from 'state/ui/utils';
+import { devtools } from 'zustand/middleware';
+import { createStore } from 'zustand/vanilla';
+
+interface threadCounterStore {
+    totalThreadsInCommunity: number;
+    totalThreadsInCommunityForVoting: number;
+    setTotalThreadsInCommunity: (name: number) => void;
+    setTotalThreadsInCommunityForVoting: (name: number) => void;
+}
+
+export const EXCEPTION_CASE_threadCountersStore = createStore<threadCounterStore>()(
+    devtools((set) => ({
+        totalThreadsInCommunity: 0,
+        totalThreadsInCommunityForVoting: 0,
+        setTotalThreadsInCommunity: (count) => set({ totalThreadsInCommunity: count }),
+        setTotalThreadsInCommunityForVoting: (count) => set({ totalThreadsInCommunityForVoting: count }),
+    }))
+);
+
+// IMPORTANT: as the name suggests, this zustand store is an exception.
+// --
+// WHAT THIS STORE DOES? -- This store store's the counts of total thread in a chain and total thread that are in
+// voting in a chain. These counters come from the /bulkOffChain api and it made sense to
+// store those counts here. Ideally we should be store thread state or any other api response state in react query.
+// Storing in zustand also makes the state 'reactive' (it will update without us having to trigger a force re-render)
+// --
+// THINGS NOT TO DO:
+// -- We should not add new state variables here, unless there is a very very very important reason to do so --
+// -- If we do add more state here, plz add a comment explaining it is added for the next person to be well aware --
+// --
+// -- THANKS :) --
```

Comment:
How can we avoid this? We are in the phase of refactoring/cleaning up the threads and this seems like a workaround aka temporary solution that usually stays around forever eventually. 

In my opinion we should extract `numTotalThreads` and `numVotingThreads` from `bulkOffchain` and handle it with react-query separately. Then, on the thread modification, we can manipulate the cache to change the number up or down. 

I understand that this is the solution for current problem, but I believe we can take a step back and refactor a bit `bulkOffchain` to not do the hacky approaches here on the frontend. 

cc @jnaviask

Reply:
+1 I totally agree. We already have a ticket to deal with that https://github.com/hicommonwealth/commonwealth/issues/3244.

In that I was going to propose this solution

- Make a new `GET /thread/counts` API that would just return the counts `numVotingThreads`, `numTotalThreads` and any other global thread count for a chain
- In the `GET /thread?bulk=true` API, we return the `total_results` field for the specified query params which would be used per topicId.

Again this is a separate thing, so that why I added the `useEXCEPTION_CASE_threadCountersStore` for the mean time (to get rid of the old thread controller)

cc: @jnaviask

Reply:
#3244 doesn't seem to imply the solution of modifying bulkOffchain. I'm wondering what the correct path would be on the API level, because I agree w Marcin that this seems like a complicated / hacky fix for what could be easily resolved at the API level. Wondering what @rbennettcw thinks...